corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -84,14 +88,41 @@ def _report_config_to_fixture(self, report_config, user):
         deferred_fields = {ui_filter.field for ui_filter in defer_filters.values()}
         filter_options_by_field = defaultdict(set)
 
-        for i, row in enumerate(data_source.get_data()):
-            row_elem = ElementTree.Element('row', attrib={'index': str(i)})
+        def _row_to_row_elem(row, index, is_total_row=False):
+            row_elem = ElementTree.Element(
+                'row',
+                attrib={
+                    'index': str(index),
+                    'is_total_row': str(is_total_row),
+                }
+            )
             for k in sorted(row.keys()):
                 value = serialize(row[k])
                 row_elem.append(self._element('column', value, attrib={'id': k}))
-                if k in deferred_fields:
+                if not is_total_row and k in deferred_fields:
                     filter_options_by_field[k].add(value)
-            rows_elem.append(row_elem)
+            return row_elem
+
+        for i, row in enumerate(data_source.get_data()):
+            rows_elem.append(_row_to_row_elem(row, i))
+
+        if data_source.has_total_row:
+            total_row = get_total_row(
+                data_source.get_data(),
+                data_source.aggregation_columns,
+                data_source.column_configs,
+                get_expanded_columns(data_source.column_configs, data_source.config)
+            )
+            rows_elem.append(_row_to_row_elem(
```

Comment:
:+1: this function seems to be growing outside of what it should ideally be responsible for